test/tests/functional/pbs_cray_vnode_per_numa.py

```diff
@@ -0,0 +1,169 @@
+        # Do not HUP now, we will do so when we reset the nodes
+        rv = self.mom.add_config({'$vnode_per_numa_node': True}, False)
+        self.assertTrue(rv)
+
+        # Start from a clean slate, delete any existing nodes and re-create
+        # them
+        momname = self.get_momname()
+        self.reset_nodes(momname)
+
+        # Get the pbsnodes -av output for comparison later
+        vnodes_pernuma = self.server.status(NODE)
+        for n in vnodes_pernuma:
+            if n['resources_available.host'] not in dncpus.keys():
+                dncpus[n['resources_available.host']] = int(
+                    n['resources_available.ncpus'])
+            else:
+                dncpus[n['resources_available.host']
+                       ] += int(n['resources_available.ncpus'])
+            if n['resources_available.host'] not in dmem.keys():
+                dmem[n['resources_available.host']] = int(
+                    n['resources_available.mem'][0:-2])
+            else:
+                dmem[n['resources_available.host']
+                     ] += int(n['resources_available.mem'][0:-2])
+            if 'resources_available.naccelerators' in n.keys():
+                if n['resources_available.naccelerators'][0] != '@':
+                    if n['resources_available.host'] not in dacc.keys():
+                        dacc[n['resources_available.host']] = int(
+                            n['resources_available.naccelerators'])
+                    else:
+                        dacc[n['resources_available.host']
+                             ] += int(n['resources_available.naccelerators'])
+
+        # Remove the configuration setting and re-read the vnodes
+        rv = self.mom.unset_mom_config('$vnode_per_numa_node', False)
+        self.assertTrue(rv)
+        self.reset_nodes(momname)
+
+        vnodes_combined = self.server.status(NODE)
+
+        # Compare the multiple vnodes values to the combined vnode output
+
+        for n in vnodes_combined:
+            if 'resources_available.PBScrayseg' in n:
+                self.logger.info(
+                    "ERROR resources_available.PBScrayseg was found.")
+                self.assertTrue(False)
+
+            self.assertEqual(int(n['resources_available.ncpus']), dncpus[
+                n['resources_available.host']])
+            self.assertEqual(int(n['resources_available.mem'][0:-2]), dmem[
+                n['resources_available.host']])
+            if 'resources_available.naccelerators' in n:
+                self.assertEqual(int(n['resources_available.naccelerators']),
+                                 dacc[n['resources_available.host']])
+
+    def restartPBS(self):
+        try:
+            svcs = PBSInitServices()
+            svcs.restart()
+        except PbsInitServicesError, e:
+            self.logger.info("PBS restart failed: \n" + e.msg)
+            self.assertTrue(e.rv)
+
+    def get_momname(self):
```

Comment:
I don't think you need this method, you can do simply get mom name by 'self.mom.shortname'
aka. self.mom is same as self.moms.values()[0]